src/py21cmfast/src/GenerateICs.c

```diff
@@ -170,6 +196,70 @@ int ComputeInitialConditions(unsigned long long random_seed, struct UserParams *
         }
     }
 
+
+    // ******* Relative Velocity part ******* //
+  if(user_params->USE_RELATIVE_VELOCITIES){
+//for now I'm going to assume NO FFTW WISDOM, until that part is fixed.
+//note we do NOT filter our boxes first, in order to keep the total number of boxes small.
+    plan = fftwf_plan_dft_c2r_3d(user_params->DIM, user_params->DIM, user_params->DIM, (fftwf_complex *)HIRES_box_vcb_x, (float *)HIRES_box_vcb_x, FFTW_ESTIMATE);
+    fftwf_execute(plan);
+    plan = fftwf_plan_dft_c2r_3d(user_params->DIM, user_params->DIM, user_params->DIM, (fftwf_complex *)HIRES_box_vcb_y, (float *)HIRES_box_vcb_y, FFTW_ESTIMATE);
+    fftwf_execute(plan);
+    plan = fftwf_plan_dft_c2r_3d(user_params->DIM, user_params->DIM, user_params->DIM, (fftwf_complex *)HIRES_box_vcb_z, (float *)HIRES_box_vcb_z, FFTW_ESTIMATE);
+    fftwf_execute(plan);
+
+// sample the UNfiltered velocity box and save it to python
+    for (i=0; i<user_params->DIM; i++){
+        for (j=0; j<user_params->DIM; j++){
+            for (k=0; k<user_params->DIM; k++){
+              vcb_x = *((float *)HIRES_box_vcb_x + R_FFT_INDEX(i,j,k));
+              vcb_y = *((float *)HIRES_box_vcb_y + R_FFT_INDEX(i,j,k));
+              vcb_z = *((float *)HIRES_box_vcb_z + R_FFT_INDEX(i,j,k));
+              boxes->hires_vcb[R_INDEX(i,j,k)] = sqrt(vcb_x*vcb_x+vcb_y*vcb_y+vcb_z*vcb_z)/VOLUME;
+            }
+        }
+    }
+  //now FFT back to complex space to filter
+    plan = fftwf_plan_dft_r2c_3d(user_params->DIM, user_params->DIM, user_params->DIM, (float *)HIRES_box_vcb_x, (fftwf_complex *)HIRES_box_vcb_x, FFTW_ESTIMATE);
+    fftwf_execute(plan);
+    plan = fftwf_plan_dft_r2c_3d(user_params->DIM, user_params->DIM, user_params->DIM, (float *)HIRES_box_vcb_y, (fftwf_complex *)HIRES_box_vcb_y, FFTW_ESTIMATE);
+    fftwf_execute(plan);
+    plan = fftwf_plan_dft_r2c_3d(user_params->DIM, user_params->DIM, user_params->DIM, (float *)HIRES_box_vcb_z, (fftwf_complex *)HIRES_box_vcb_z, FFTW_ESTIMATE);
+    fftwf_execute(plan);
+
+  //and filter each box:
+    if (user_params->DIM != user_params->HII_DIM){
+      filter_box(HIRES_box_vcb_x, 0, 0, L_FACTOR*user_params->BOX_LEN/(user_params->HII_DIM+0.0));
+      filter_box(HIRES_box_vcb_y, 0, 0, L_FACTOR*user_params->BOX_LEN/(user_params->HII_DIM+0.0));
+      filter_box(HIRES_box_vcb_z, 0, 0, L_FACTOR*user_params->BOX_LEN/(user_params->HII_DIM+0.0));
+    }
+
+  //to save into a lowres python box
```

Comment:
this is not really a "python" box -- it's a C struct. But it is the thing we can easily access in python :-)